metrics/stackdriver_exporter.go

```diff
@@ -63,10 +75,32 @@ func newOpencensusSDExporter(o stackdriver.Options) (view.Exporter, error) {
 // TODO should be properly refactored to be able to inject the getMonitoredResourceFunc function.
 // 	See https://github.com/knative/pkg/issues/608
 func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	gm := gcpMetadataFunc()
+	gm := getGCPMetadata(config)
 	mtf := getMetricTypeFunc(config.stackdriverMetricTypePrefix, config.stackdriverCustomMetricTypePrefix)
+
+	proj := config.stackdriverProjectID
+	if config.stackdriverConfig.ProjectID != "" {
+		proj = config.stackdriverConfig.ProjectID
+	}
+
```

Comment:
Why not use `gm.project` directly and get rid of this part?

Reply:
The if-clause here is to ensure "config-stackdriver" will take precedent over "config-observability" or "config-logging". (though admittedly it is a little hard to read).

I can move this logic to getGCPMetadata though.